api/v1alpha1/catalogentry_types.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CatalogEntrySpec defines the desired state of CatalogEntry
+type CatalogEntrySpec struct {
+	// The workspace name where the APIExport locates
+	Workspace string `json:"workspace,omitempty"`
+	// The name of the APIExport
+	ExportName string `json:"exportName,omitempty"`
+	// A list of icon data that can be displayed in UI
+	// +optional
+	Icons []Icon `json:"icons,omitempty"`
```

Comment:
what do multiple icons means?

Reply:
the first compatible icon to be displayed, e.g. by media type?

Reply:
I modeled this after CSV on OLM project to allow multiple icons with different types and the UI can choose from depending on what they need. However, if this seems excessive, I'm happy to keep it singular.

Reply:
my comment was mostly about carity in the API docs. How is this list to be understood. I.e. which does the order mean.

Reply:
Looks like this is singular now